### PR TITLE
Issue #7145: aligned javadoc/xdoc for NPathComplexity

### PR DESCRIPTION
**Pull request by pbludov**
Issue #7145 (part of #5750)

No changes in the code.

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/NPathComplexityCheck.java [hunk 1]
```diff
@@ -29,12 +29,101 @@
 import com.puppycrawl.tools.checkstyle.api.TokenTypes;
 
 /**
- * Checks the npath complexity against a specified limit (default = 200).
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/NPathComplexityCheck.java]: I think the description is better with this as the first sentence, minus the default.

Comment (pbludov) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/NPathComplexityCheck.java]: done.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/NPathComplexityCheck.java [hunk 2]
```diff
@@ -28,13 +28,104 @@
 import com.puppycrawl.tools.checkstyle.api.DetailAST;
 import com.puppycrawl.tools.checkstyle.api.TokenTypes;
 
-/**
- * Checks the npath complexity against a specified limit (default = 200).
- * The npath metric computes the number of possible execution paths
- * through a function. Similar to the cyclomatic complexity but also
- * takes into account the nesting of conditional statements and
- * multi-part boolean expressions.
+/**<p>
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/NPathComplexityCheck.java]: Shouldn't the `<p>` start on the next line?

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/NPathComplexityCheck.java]: I have no strict preference

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/NPathComplexityCheck.java]: In PR we have a new Check that will enforce strict layout.

Comment (pbludov) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/NPathComplexityCheck.java]: done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/NPathComplexityCheck.java [hunk 3]
```diff
@@ -29,12 +29,105 @@
...
+ * </blockquote>
+ * <blockquote>
+ * Although strategies to reduce the NPATH complexity of functions are important,
+ * care must be taken not to distort the logical clarity of the software by
+ * applying a strategy to reduce the complexity of functions. That is, there is
+ * a point of diminishing return beyond which a further attempt at reduction of
+ * complexity distorts the logical clarity of the system structure.
+ * </blockquote>
+ * <table>
+ * <caption>Examples</caption>
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/NPathComplexityCheck.java]: This is new.
@pbludov Can you explain this addition after approvals?

Comment (pbludov) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/NPathComplexityCheck.java]: javadoc lint wants caption or summary for each table:

https://stackoverflow.com/questions/22528767/how-to-work-around-the-stricter-java-8-javadoc-when-using-maven

**Pull request merged**